src/spikeinterface/core/analyzer_extension_core.py

```diff
@@ -80,7 +81,7 @@ def _select_extension_data(self, unit_ids):
     def _get_data(self):
         return self.data["random_spikes_indices"]
 
-    def some_spikes(self):
+    def random_spikes(self):
```

Comment:
The name some_spikes was very fuzzy now this `random_spikes()` is becoming total cryptic.
This sounds like "make the randomization".
The idea of this function is "give and cache it the sub spikes vector that correspond to the sampling."
So the is function for me is even worst that the previous one which was already bad.

Reply:
What about `get_sampled_spikes`?

Reply:
Maybe yes.

Reply:
Or `sample_random_spikes` because you are "sampling" your pool of random spikes. I think this is what Heberto convinced me of. Unless we change the overall function to `sample_random_spikes`, but again my mental schema is that each class is an object (we don't call it "CalculateSpikeAmplitudes" we call it "SpikeAmplitudes" so the object is `RandomSpikes` for me and what we do is compute them. Then other functions sample those random spikes later. Again that's just my mental model of this.

Reply:
Here this method is not making the sampling which is done in the   `_run()`.
This method is making the slice of the spike vector and caching it!!

Reply:
In that case since this is just a normal extension shouldn't it just be the `get_data` then? So the whole api is consistent?

Reply:
The get_data() give the indicies this method give the spike vector sliced by this indicies but cached because we acces this slice a lot

Reply:
I finally (I hope) understand what's going on. In that case this is really
`get_cached_sampled_spikes` or `get_cached_random_spikes`.  I think saying this is going to the cached spikes is clearer because if we recompute the random spikes we should remake the cache so it should be clear that we are grabbing from the cache and that no additional calculation should be occurring.

Reply:
What about just `get_sampled_spikes()`? Caching is something we do internally to speed things up, so I don't think it makes sense to have it in the name

Reply:
I think I like that name (get_sampled_spikes) the best, but would you then want to change the class/extension to `SampledSpikes` to be consistent? Again for me at least `RandomSpikes` has an intuitive flavor even if "sampling" is the more rigorous term.

Reply:
I like random too. I'll co for `get_random_spikes()` then for consistency